Automattic/wp-calypso
# Add upgrade click handlers to the plan page

Task-related: https://github.com/Automattic/wp-calypso/issues/68089

Figma file: HOg65lHD0QOfaq0QtzKQbC-fi-507%3A10329

### Proposed Changes

* This PR reuses the CTAButton to handle clicks on the plugins/plans page.

Before | After
--|--
![before-plugin-plans](https://user-images.githubusercontent.com/140841/192055438-0340f2b5-2c7a-4663-8827-6aab9e80dc90.png)  | ![after-plugin-plans](https://user-images.githubusercontent.com/140841/192055454-4c9b2eed-4ada-4cc1-bdf4-33d2c1614319.png)

### Acceptance criteria

- [ ] CTAButton should accept the new optional `desiredPlan` [parameter](https://github.com/Automattic/wp-calypso/pull/68154/files#diff-d5ac7cfa2d86b94db92445b2f1e3f311cafbe173848e5b2cc4dd86d550f90ae6R272): So we can redirect users to business or eCommerce plan
- [ ] CTAButton should accept the new optional `buttonText` [parameter](https://github.com/Automattic/wp-calypso/pull/68154/files#diff-d5ac7cfa2d86b94db92445b2f1e3f311cafbe173848e5b2cc4dd86d550f90ae6R113): So we can customize its text according to Figma
- [ ] CTAButton should accept the new optional `buttonClassName` [parameter](https://github.com/Automattic/wp-calypso/pull/68154/files#diff-d5ac7cfa2d86b94db92445b2f1e3f311cafbe173848e5b2cc4dd86d550f90ae6R177): So we can customize its classes depending on which screen uses it
- [ ] In case the plans page is opened with a given plugin, [we should render the CTAButton as CTA](https://github.com/Automattic/wp-calypso/pull/68154/files#diff-d89d039d749332c2d6c1c257980ee3273b0db6689a013b93a7abf58a9e6267f1R192)
- [ ] In case the plans page is opened without plugins, from the Upsell nudge, [we should render a simple button as CTA](https://github.com/Automattic/wp-calypso/pull/68154/files#diff-d89d039d749332c2d6c1c257980ee3273b0db6689a013b93a7abf58a9e6267f1R204)
- [ ] In case we have eligibility warning messages, and we already are at the new plans page, [we don't need to redirect the user to the plugin plans page](https://github.com/Automattic/wp-calypso/pull/68154/files#diff-d5ac7cfa2d86b94db92445b2f1e3f311cafbe173848e5b2cc4dd86d550f90ae6R131).


### Testing Instructions

- Open the Calypso plugins page [with the flag parameter](http://calypso.localhost:3000/plugins/?flags=plugins-plans-page)
- Select a simple site on a free plan
- Open a given plugin and try to install it
- You should see the new /plugins/plans page
- The actions button should redirect you to the checkout after passing through eligibility warnings
- If you click at the upsell nudge on /plugins page, you should see the new /plugins/plans page and be redirected to checkout without seeing eligibility warnings messages

#### Pre-merge Checklist






Fixes #68089
